src/project.cc

```diff
@@ -264,8 +261,9 @@ std::vector<Project::Entry> LoadCompilationEntriesFromDirectory(
     CompileCommandsEntry entry;
     entry.args.reserve(num_args);
     for (unsigned j = 0; j < num_args; ++j) {
-      entry.args.push_back(
-          ToString(clang_CompileCommand_getArg(cx_command, j)));
+      std::string arg = ToString(clang_CompileCommand_getArg(cx_command, j));
+      if (arg != relative_filename)
```

Comment:
Can you move this check to GetCompilationEntryFromCompileCommandEntry? Then we can enable the #if false test and have this logic unit-tested. It also means it can be more easily used when loading projects not via compile_commands.json.

Reply:
We may also want to normalize paths before checking.

Reply:
Changed.

This should be the minimal change to make relative path pass. blacklist stuff is not touched.